internal/fsutil/create_test.go

```diff
@@ -73,18 +75,58 @@ func (s *S) TestCreate(c *C) {
 	}()
 
 	for _, test := range createTests {
+		if test.result == nil {
+			// Empty map for no files created.
+			test.result = make(map[string]string)
+		}
 		c.Logf("Options: %v", test.options)
 		dir := c.MkDir()
 		options := test.options
 		options.Path = filepath.Join(dir, options.Path)
-		err := fsutil.Create(&options)
+		fsCreator := fsutil.NewCreator()
+		err := fsCreator.Create(&options)
 		if test.error != "" {
 			c.Assert(err, ErrorMatches, test.error)
-			continue
 		} else {
 			c.Assert(err, IsNil)
 		}
-		result := testutil.TreeDump(dir)
-		c.Assert(result, DeepEquals, test.result)
+		c.Assert(testutil.TreeDump(dir), DeepEquals, test.result)
+		if test.options.MakeParents {
+			// The fsCreator does not record the parent directories created
+			// implicitly.
+			for path, info := range treeDumpFSCreator(fsCreator, dir) {
+				c.Assert(info, Equals, test.result[path])
+			}
+		} else {
+			c.Assert(treeDumpFSCreator(fsCreator, dir), DeepEquals, test.result)
+		}
+	}
+}
+
+func treeDumpFSCreator(cr *fsutil.Creator, root string) map[string]string {
```

Comment:
Needs a comment explaining the delta from testutil.TreeDump.

Reply:
It uses the same format for outputs but instead of traversing the filesystem uses the data inside the ` fsutil.Creator`.

Reply:
Added comment in code.